addStyles.js

```diff
@@ -13,8 +13,16 @@ var stylesInDom = {},
 	isOldIE = memoize(function() {
 		return /msie [6-9]\b/.test(window.navigator.userAgent.toLowerCase());
 	}),
-	getHeadElement = memoize(function () {
-		return document.head || document.getElementsByTagName("head")[0];
+	getElement = (function(fn) {
+		var memo = {};
+		return function(selector) {
+			if (typeof memo[selector] === "undefined") {
+				memo[selector] = fn.call(this, selector);
+			}
+			return memo[selector]
+		};
+	})(function (styleTarget) {
+		return document.querySelector(styleTarget)
```

Comment:
Do we really? According to Can I Use, the support for [querySelector](http://caniuse.com/#search=querySelector) in newer browsers is better than that of [getElementsByClassName](caniuse.com/#search=getElementsByClassName), the latter not having any support for IE8 while the former has partial support.